### DISCUSSION TITLE
Include package data?

### DISCUSSION THREAD

**Discussion by pawamoy** [pdm-project/pdm#569] [Q&A] [answered]
What's the recommended way to add arbitrary files as package data into the sdists and wheels?
[Includes](https://pdm.fming.dev/pyproject/tool-pdm/#include-and-exclude-package-files)?
Are these includes properly exported as `package_data` when generating the setup.py?

**Comment by pawamoy** [comment, marked as answer]
Ah, it seems that files in the folder used for `package-dir` are recursively included by default. Good enough for me.

**Reply by pawamoy** [reply]
Coming back to this. It seems this behavior is not reflected by the setup.py generated by PDM (1.15.3).
When I do `pdm build`, my data files are correctly added to both sdists and wheels.
But when I run `python setup.py sdist bdist_wheel`, the data files are not included.
The setup.py file is generated with `pdm export -f setuppy`.

**Reply by frostming** [reply]
Yes, setuptools has a much more convoluted system to decide whether to include or exclude a certain file. PDM doesn't plan to support that. Users are suggested to write `MANIFEST.in` themselves if they want to use `pdm export -f setup.py` for distribution.

**Reply by pawamoy** [reply]
Fair enough, thank you for the explanation :) I simply added `__init__.py` modules where it matters so that setuptools picks up my data folders automatically.
